src/genegraph/source/graphql/schema/variation_descriptor.clj

```diff
@@ -158,8 +151,24 @@
         {variation-iri :variation_iri} args
         variation-resource (q/resource variation-iri)
         ;descriptor-resources (q/ld-> variation-resource [[:vrs/xrefs :<]])
-        descriptor-resources (q/select "SELECT ?descriptor WHERE { ?descriptor :vrs/xrefs ?variation_iri }"
-                                       {:variation_iri (str variation-resource)})
+
+
+        ;by-iri (q/select "SELECT ?iri WHERE { ?iri a :vrs/CategoricalVariationDescriptor }"
+        ;                 {:iri (str variation)})
+        resource-has-type? (fn [r class-kw]
+                             (let [type-qualified (str (get names/local-class-names class-kw))]
+                               (log/debug :fn :resource-has-type? :r r :class-kw class-kw :type-qualified type-qualified)
+                               (let [hasit? (not-empty (q/select "SELECT ?iri WHERE { ?iri a ?type }"
+                                                                 {:iri r
+                                                                  :type (q/resource type-qualified)}))]
+                                 (log/debug :fn :resource-has-type? :hasit? hasit?)
+                                 hasit?
+                                 )))
+        descriptor-resources
+        (if (resource-has-type? variation-resource :vrs/CategoricalVariationDescriptor)
+          [variation-resource]
+          (q/select "SELECT ?descriptor WHERE { ?descriptor :vrs/xrefs ?variation_iri }"
+                    {:variation_iri (str variation-resource)}))
```

Comment:
`hasit? hasit?` is funny.
Isn't `type-qualified` just `(class-kw names/local-class-names)`?